File: LeetCode/merge_sorted_array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        vector<int> aux;
        
        int i, j;
        i = j = 0;
        while(i < m && j < n) {
            if(nums1[i] <= nums2[j])
                aux.push_back(nums1[i++]);
            else
                aux.push_back(nums2[j++]);
        }
        
        while(i < m)
            aux.push_back(nums1[i++]);
        while(j < n)
            aux.push_back(nums2[j++]);
        
        for(int k = 0; k < nums1.size(); k++) {
            nums1[k] = aux[k];
        }
    }
};

File: LeetCode/merge_sorted_array.go
func merge(nums1 []int, m int, nums2 []int, n int)  {
    i, j, last := m-1, n-1, m+n-1

    for j >= 0 {
        if i >= 0 && nums1[i] > nums2[j] {
            nums1[last] = nums1[i]
            i--
        } else {
            nums1[last] = nums2[j]
            j--
        }
        last--
    }
}
